### PR TITLE
Update conduit_fortran_obj.f90 to initialize node pointer to NULL.

### PR DESCRIPTION
**Pull request by aaroncblack**
Small change to initialize the cnode pointer to null.

GNU is not always giving the correct result of a
c_associated(cnode)

when that isn't set to null beforehand.

### CODE REVIEW

File: src/libs/conduit/fortran/conduit_fortran_obj.f90 [hunk 1]
```diff
@@ -10,12 +10,12 @@
...
 !------------------------------------------------------------------------------
     use conduit
-    use, intrinsic :: iso_c_binding, only : C_PTR
+    use, intrinsic :: iso_c_binding, only : C_PTR, C_NULL_PTR
     implicit none
     
     !--------------------------------------------------------------------------
     type node
-        type(C_PTR) cnode
+        type(C_PTR) cnode => C_NULL_PTR
```
Comment (aaroncblack) [inline src/libs/conduit/fortran/conduit_fortran_obj.f90]:
```suggestion
        type(C_PTR) :: cnode = C_NULL_PTR
```
```suggestion
        type(C_PTR) cnode => C_NULL_PTR
```

**Comment by cyrush** [comment]
Compiler still not happy :-(

Are there example patterns we can follow for this, say in something like shroud?

**Comment by cyrush** [comment]
@aaroncblack is this still undermining use for you? Any place we can look for an example of how to avoid the gfortran issue?

**Comment by aaroncblack** [comment]
I'll close this.  I can work around it by creating a separate bool 'is_initialized' that I check, or move the code to C++, etc.

**Comment by cyrush** [comment]
thanks @aaroncblack -- if it appears in other contexts we can revisit.

**Pull request closed without merging**